### PR TITLE
Implement check for alphanumericals for findTag, resolve bugs, and update UG

### LINKED ISSUE #139: Bug Report: findTag t/TagName
Typing the above into GUI command line results in an IllegalArgumentException that does not display any message on GUI

### LINKED ISSUE #153: [PE-D] Try to use find command with multiple words and somehow works
In the documentation, it is written that only a KEYWORD can be used. When testing using multiple KEYWORDS, all tasks related to ONE of the keywords are shown.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: austenjs/ped#6

### LINKED ISSUE #155: [PE-D] Nothing happen when using findTag with input a symbol
Try to findTag with the parameter '. Nothing happened when press and no error shown.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: austenjs/ped#8

### LINKED ISSUE #193: [PE-D] find & findTag: Inconsistency in case sensitivity
The `find` command is case insensitive while `findTag` is case-sensitive. It would be best if it was consistent, or if not, then it would be clearer if this note was reflected in the user guide.

`find project` shows results for both project and Project:

`findAppt highpriority` shows results for highpriority but not highPriority:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: edelinetenges/ped#7

### LINKED ISSUE #199: [PE-D] Add command error message for invalid module code is not accurate
Command entered: `add n/TP v1.2 a/2021-01-30 10:00 b/2021-01-30 12:00 m/CS103T d/Finish basic commands for TP w/1 r/monthly t/highPriority`

> 2. The {MAJOR_NAME} should only contain 2 or 3 capital letter.

According to the UG, all the supported modules only have 2 letters as the major name, there isn't any that is 3. Furthermore, there is a grammatical error where "letter" should have been "letters".

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: kouyk/ped#1

### LINKED ISSUE #201: [PE-D] Wrong error message when entering invalid module code for `mod` command
Example: `mod CS2103`

Error message:
```
Invalid command format! 
mod: Find tasks associated with the Module code provided and displays them as a list with index numbers.
Parameters: MODULE_CODE 
Example: mod CS2103T
```

In this case I believe that my command format is correct, the thing that is wrong is the unsupported module code. In this case the error message should be informing me about the invalid module code and tell me what the constraints are the for module code. Or else I will be confused and repeatedly verify my command to check if there is missing prefix etc.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: kouyk/ped#3

### LINKED ISSUE #207: [PE-D] Inconsistent case-sensitivity among find commands
`find` command is case-insensitive.
`findTag` is case-sensitive.

`findTag` should also be case-insensitive. Either that or make it explicit in the UG that tags are case-sensitive throughout the application.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: kouyk/ped#9

### LINKED ISSUE #208: [PE-D] `find` and `findTag` commands accepts more than a single keyword
From the UG since `KEYWORD` is singular, it is assumed that the application will only accept a single keyword or that multiple keywords are treated as a single keyword as-is.

Example:
The task list contains a task called "Team Project", `find team` and `find project` both works as expected. However, it is surprising to see that `find random project` works as well. It can be inferred that multiple keywords are split by space and the tasks will show up as long as its name contains one of the keywords. This behaviour is not what is stated in the UG.

The same bug is applicable to the `findTag` command as well.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: kouyk/ped#10

### PR DESCRIPTION
**Pull request by figo2127**

### CODE REVIEW

File: src/main/java/seedu/module/model/ModuleManager.java [hunk 1]
```diff
@@ -24,8 +24,10 @@
     private static HashMap<Module, Integer> moduleMediumWorkLoadDistribution;
     private static HashMap<Module, Integer> moduleHighWorkLoadDistribution;
     private static List<String> supportedModulesInStr;
-    private static final String[] arrOfModules = {"CS1101S", "CS1231S", "CS2030", "CS2040S", "CS2101",
-        "CS2103T", "CS2105", "CS2106", "CS3230", "CS3243", "CS3244", "IS1103", "ST2131"};
+    private static final String[] arrOfModules = {"CS1010S", "CS1101S", "CS1231S",
```
Comment (yhtMinceraft1010X) [inline src/main/java/seedu/module/model/ModuleManager.java:27]: I think the new modules should also be included in the UG table

**Review by yhtMinceraft1010X — COMMENTED** [review]
Generally LGTM, apart from updating of mods in UG

File: src/main/java/seedu/module/model/ModuleManager.java (hunk 1 above)
Comment (figo2127) [reply src/main/java/seedu/module/model/ModuleManager.java:27]: Yep included them. I went out for breakfast halfway lol

File: src/main/java/seedu/module/logic/commands/FindTagCommand.java [hunk 2]
```diff
@@ -60,11 +62,35 @@ public CommandResult execute(Model model) throws CommandException {
...
+        //convert tags of Task into strings
+        for (Tag t : tags) {
+            strTagsInLower.add(t.tagName.toLowerCase());
+        }
+        String target = tag.tagName.toLowerCase();
+        if (strTagsInLower.contains(target)) {
+            return true;
+        } else {
+            return false;
+        }
```
Comment (figo2127) [inline src/main/java/seedu/module/logic/commands/FindTagCommand.java:91]: seems like I can do this in 1 line. Will change it later!

**Review by figo2127 — COMMENTED** [review]
[]

File: src/main/java/seedu/module/ui/TaskListPanel.java [hunk 3]
```diff
@@ -62,7 +62,7 @@ protected void updateItem(Task task, boolean empty) {
                 } else if (colorToAssign.equals("Yellow")) {
                     addedTaskCard.getRoot().setStyle("-fx-background-color: #F0E68C; ");
                 } else {
-                    addedTaskCard.getRoot().setStyle("-fx-background-color: #DC143C; ");
+                    addedTaskCard.getRoot().setStyle("-fx-background-color: #FF6347; ");
```
Comment (yhtMinceraft1010X) [inline src/main/java/seedu/module/ui/TaskListPanel.java:65]: I think the UG `Features` section should also be updated to describe the task colours.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -294,7 +307,9 @@ Examples:
 
 ### Search tasks with tag: `findTag`
 
-Searches for tasks with an associated tag. Note that tags are case-sensitive.
+Searches for tasks with an associated tag. Tag is case-insensitive for your convenience.
+
+One Tag allowed. Tag provided mut be alphanumerical.
```
Comment (yhtMinceraft1010X) [inline docs/UserGuide.md]: Noticed a spelling mistake `mut` instead of `must`.

Comment (figo2127) [reply docs/UserGuide.md]: oops! will change now

**Review by AiwassPrime — COMMENTED** [review]
LGTM

**Pull request merged**